lib/hammer_cli/context.rb

```diff
@@ -3,9 +3,10 @@
 module HammerCLI
 
   def self.context
-    {
+    @context ||= {
       :defaults => HammerCLI.defaults,
-      :is_tty? => HammerCLI.tty?
+      :is_tty? => HammerCLI.tty?,
+      :api_connection => HammerCLI::Connection.new(Logging.logger['Connection'])
```

Comment:
I'm not sure I understand. The context is passed to commands' constructors and is accessible as `self.context` or `@context`.
I don't think setting a default value for the api connection makes sense in this case, because we need one connection for the whole hammer.